misc/helper-scripts/rosmsg_to_asn1.py

```diff
@@ -0,0 +1,417 @@
+        msg_string += " SEQUENCE (SIZE(0.." + str(lengths[i])+")) OF "
+        msg_string += slot_asn1_types[i] + "\n"
+        field_types[i] = "simple"
+        slot_asn1_types[i] = "L" + slots[i].replace("_", "-")
+
+    for i in variable_indices:
+        msg_string += "V" + slots[i].replace("_", "-") + "::="
+        msg_string += " SEQUENCE (SIZE(0..256)) OF "
+        msg_string += slot_asn1_types[i] + "\n"
+        field_types[i] = "simple"
+        slot_asn1_types[i] = "V" + slots[i].replace("_", "-")
+
+    if (num_fields == 1):
+        if field_types[0] == "simple":
+            if slots[0] == "type":
+                slots[0] += "-" + slot_asn1_types[0]
+            if slots[0].lower() == slot_asn1_types[0].lower():
+                slots[0] += "-field"
+            msg_string += msg_name + "::=\nSEQUENCE\n{\n"
+            msg_string += "\t" + slots[0].replace("_", "-") + "\t"
+            msg_string += slot_asn1_types[0] + "\n}\n"
+    else:
+        msg_string += msg_name + "::=\nSEQUENCE\n{\n"
+        comma = ""
+        for i in range(num_fields):
+            if field_types[i] == "simple":
+                if slots[i] == "type":
+                    slots[i] += "-" + slot_asn1_types[i]
+                if slots[i].lower() == slot_asn1_types[i].lower():
+                    slots[i] += "-field"
+                msg_string += comma + "\t" + slots[i].replace("_", "-")
+                msg_string += "\t" + slot_asn1_types[i]
+                comma = ",\n"
+        msg_string += "\n}\n"
+    msg_string += "END"
+
+    return msg_string, msg_deps
+
+
+def save_msg(asn1_str, msg, out_dir):
+    """
+    Save the ASN.1 generated message to a file.
+    :param asn1_str: The message string.
+    :param msg: The message name.
+    :param out_dir: The output directory.
+    """
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+    if "/" in msg:
+        msg = msg.split("/")[1]
+    f = open(out_dir + msg + ".asn", "w")
+    f.write(asn1_str)
+    f.close()
+
+
+def main():
+    # Get the messages to be generated
+    args = parse_args()
+    print(args)
+    messages = args.messages
+    out_dir = args.output
+    if out_dir[-1] != "/":
+        out_dir += "/"
+
+    log_formatter = logging.Formatter(
+        "%(asctime)s [%(threadName)-12.12s] [%(levelname)-5.5s]  %(message)s")
+    root_logger = logging.getLogger()
+    if args.verbose:
+        root_logger.setLevel(logging.DEBUG)
+
+    if args.log:
+        log_file = args.logfile
+        if log_file is None:
+            log_file = "/tmp/rosmsg_to_asn1.log"
+        file_handler = logging.FileHandler(log_file)
+        file_handler.setFormatter(log_formatter)
+        if args.verbose:
+            file_handler.setLevel(logging.DEBUG)
+        root_logger.addHandler(file_handler)
+
+    console_handler = logging.StreamHandler(sys.stdout)
+    if args.verbose:
+        console_handler.setLevel(logging.DEBUG)
```

Comment:
Can this be avoided by applying it once after adding all the handlers?

Reply:
AFAIK each of the handlers has to have its level set.

Reply:
OK.